Fix for issue 2994 - unable to run JUnit tests locally

LeoNerdoG opened the pull request:
I have changed the category for running pure JUnit tests. Before this user could only run
the tests with command "mvn test -Dgroups='org.eclipse.kapua.test.junit.JUnitTests'", but
this changed, when this category got moved to qa/markers/...
Instead of using this default category for running JUnit tests, I have added a profile ("junitTests")
that enable developers and testers to run tests in Terminal on local machine with command:
"mvn test -PjunitTests".

This is the same as running tests with command "mvn test -Dgroups='org.eclipse.kapua.qa.markers.Categories$junitTests'",
but is much shorter and therfore easier to executed.
I recommend running this command prior to every push/PR to see, if everything is OK.

I have added ony one category in "qa/markers" module ("junitTests"), but I will also add one for running cucumber unit
tests and one for running cucumber integration tests later.

I have deleted a "unit" test that has been in "console" module for a long period of time, but it does nothing
and should actually not be there anyway:
deleted kapua/console/module/api/src/test/java/org/eclipse/kapua/app/console/module/api/test/SplitTooltipStringTest.java

POMs changed:
 - root kapua pom.xml
 - qa/integration/pom.xml
 - github actions yml file

**Related Issue**
this fixes issue #2994.

**Description of the solution adopted**
This is a fairly simpel solution, I only modified the Category for running unit tests, POMs (added "junitTests" profile) and GHA yaml file, becasuse category has been changed. 

**Any side note on the changes made**
RunTests.md file has also been updated (with new command and some minor syntax errors).

@nezau-ct , @sonja-ct Can you please checkout this branch and run the tests locally on your machines with command "mvn test-PjunitTests" and report if the results are OK? Tests should be running for a couple of minutes (on my machine it takes 5 minutes, it may take more on your machines). 

@Coduz can you please check the PR and merge it when you have time? Thanks! 

Signed-off-by: Leonardo Gaube <leonardo.gaube@endava.com>

Coduz commented on broker/core/src/test/java/org/eclipse/kapua/broker/core/plugin/DefaultConnectorDescriptionProviderTest.java:
```diff
@@ -12,32 +12,36 @@
+    public void cleanUp() {
+        System.clearProperty("broker.connector.descriptor.default.disable");
+        System.clearProperty("broker.connector.descriptor.configuration.uri");
+        System.out.println("broker-default-disable = " + System.getProperty("broker.connector.descriptor.default.disable"));
```
Please use SLF4J logger or just remove them.

Same for all `System.out.println`

Coduz commented on service/security/shiro/src/test/java/org/eclipse/kapua/service/authentication/shiro/utils/AuthenticationUtilsTest.java:
```diff
@@ -14,26 +14,35 @@
         plainValues = new String[]{"plain_..val9&^%ue123!!", "   value#999 ?><,,..;''a  ", "valu   e plain*&^%     $#45", "value,,,,va?><  ", "... s_er%%67nsaa4356&^%   a *(me"};
         encryptedValues = new String[]{"2c3mAagxwaEuAhmR1UzyafpKdA8R-poaS2upJPj4kzE", "QprB8vCeyft4pU8AJdxSWlIFL1b02s-UqTQwirKj9Dw", "RrRtzYPLFDgVdmKo9kOipZv723WBs2J3IxSoPwSJM7g",
                 "gcGjWNELoVl9R-71-Nm8aAoNgf3lxr5FziYhj8dmML0", "lCysXXE00k64hm_FzQ8aK1GlVMFqR6So3knfnb5R_CQKDYH95ca-Rc4mIY_HZjC9"};
+        try (InputStream input = new FileInputStream("/Users/leonardo/kapua/service/security/shiro/src/main/resources/kapua-crypto-setting.properties")) {
```
This must be removed?

Coduz reviewed (changes requested):
There are some things to change and also why there are plenty of test that have been removed?

LeoNerdoG commented:
Hey @Coduz,
I have resolved the two things you commented. 
The reaseon, why I deleted some tests are:
1. They are somehow duplicated and could be deleted or
2. System.setProperty was not working "properly" and we had failing tests when running in the terminal. 

The problem is, that IntelliJ has a different runner than e.g. GHA or Terminal. Because of this, some of the tests were failing when running locally (but not failing when running on Travis, GHA, IntelliJ or Eclipse). By removing them, we are not losing anything, as they should not be there in the first place (tests should always be successfull, no matter how they are run). 

If this is not OK, I can try and find a different solution.

Commit:
Fix for issue 2994 - unable to run JUnit tests locally

I have changed the category for running pure JUnit tests. Before this user could only run
the tests with command "mvn test -Dgroups='org.eclipse.kapua.test.junit.JUnitTests'", but
this changed, when this category got moved to qa/markers/...
Instead of using this default category for running JUnit tests, I have added a profile ("junitTests")
that enable developers and testers to run tests in Terminal on local machine with command:
"mvn test -PjunitTests".

This is the same as running tests with command "mvn test -Dgroups='org.eclipse.kapua.qa.markers.Categories$junitTests'",
but is much shorter and therfore easier to executed.
I recommend running this command prior to every push/PR to see, if everything is OK.

I have added ony one category in "qa/markers" module ("junitTests"), but I will also add one for running cucumber unit
tests and one for running cucumber integration tests later.

I have deleted a "unit" test that has been in "console" module for a long period of time, but it does nothing
and should actually not be there anyway:
deleted kapua/console/module/api/src/test/java/org/eclipse/kapua/app/console/module/api/test/SplitTooltipStringTest.java

POMs changed:
 - root kapua pom.xml
 - qa/integration/pom.xml
 - github actions yml file

RunTests.md file has also been updated (with new command and some minor syntax errors).

Signed-off-by: Leonardo Gaube <leonardo.gaube@endava.com>

nezau-ct commented:
I have run it on Ubuntu 18.04.5 and the tests finished successfully.
